### PR TITLE
feat: add navigate method

### PR DESCRIPTION
**Pull request by jhutchins**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by lucasfernog** [comment]
I think we can target this for the `1.x` branch instead so it lands on v1.5.

File: core/tauri/src/window.rs [hunk 1]
```diff
@@ -1648,8 +1648,14 @@ impl<R: Runtime> Window<R> {
...
+  /// Navigates the webview to the defined url.
+  #[cfg(not(test))]
+  pub fn navigate(&mut self, url: Url) {
+    self.window.dispatcher.navigate(url).unwrap();
+  }
+
   #[cfg(test)]
-  pub(crate) fn navigate(&mut self, url: Url) {
+  pub fn navigate(&mut self, url: Url) {
     self.current_url = url;
```
Comment (lucasfernog) [inline core/tauri/src/window.rs]: Now that we have an actual `navigate` function, we can remove the [current_url field](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/window.rs#L782) and implement `fn navigate` under the [mock runtime](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/test/mock_runtime.rs#L553). Basically you'll need to change the [Dispatcher#url](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/test/mock_runtime.rs#L193) to a `Mutex<String>` and mutate it on the `navigate` impl.

With this change we have less `#[cfg(test)] #[cfg(not(test))]` to deal with.

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: core/tauri/src/window.rs (hunk 1 above)
Comment (jhutchins) [reply core/tauri/src/window.rs]: I'll get that added.

Comment (jhutchins) [reply core/tauri/src/window.rs]: I've reworked how url is handled in test contexts.

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
Perfect!

**Comment by kanatapple** [comment]
@lucasfernog 
Will this feature ship in v1.5?
If so, when will v1.5 be released?

**Pull request merged**